templates/repo/issue/view_content/add_reaction.tmpl

```diff
@@ -1,13 +1,14 @@
 {{if .ctxData.IsSigned}}
 <div class="item action ui dropdown jump pointing top right select-reaction" data-action-url="{{.ActionURL}}">
-	<a class="add-reaction">
+	<a class="add-reaction muted">
 		{{svg "octicon-smiley"}}
 	</a>
 	<div class="menu">
-		<div class="header">{{.ctxData.locale.Tr "repo.pick_reaction"}}</div>
+		<div class="header reactions-header">{{.ctxData.locale.Tr "repo.pick_reaction"}}</div>
 		<div class="divider"></div>
 		{{range $value := AllowedReactions}}
-			<a class="item reaction" data-tooltip-content="{{$value}}" data-reaction-content="{{$value}}">{{ReactionToEmoji $value}}</a>
+			{{/* the popup menu is only used for "adding reaction", so do not put data-has-reacted on it */}}
```

Comment:
I disagree, this menu should also enable to un-react and that is also how it works on Github. I had this already fixed but you "broke" it again.

Reply:
I didn't, that's your logic and I just added that comment

Reply:
If you want to make it support unreact, just add that attribute, I think that's enough because the JS shares the same logic

Reply:
Fixed it in https://github.com/go-gitea/gitea/pull/24760/commits/591ffb0a37b06546dcdd1b385ea5769b0ff73fe4 with a complex selector, but I will try your method.

Reply:
Adding `hasReacted` to that template requires more backend changes because that template uses `AllowedReactions` which does not have the metadata about the user, so I'll leave my solution.